disnake/auto_moderation.py

```diff
@@ -0,0 +1,495 @@
+    """
+
+    __slots__ = ("type", "_metadata")
+
+    @overload
+    def __init__(self, *, type: Literal[AutomodActionType.block_message]):
+        ...
+
+    @overload
+    def __init__(self, *, type: Literal[AutomodActionType.send_alert_message], channel: Snowflake):
+        ...
+
+    @overload
+    def __init__(
+        self,
+        *,
+        type: Literal[AutomodActionType.timeout],
+        timeout_duration: Union[int, timedelta],
+    ):
+        ...
+
+    def __init__(
+        self,
+        *,
+        type: AutomodActionType,
+        channel: Optional[Snowflake] = None,
+        timeout_duration: Optional[Union[int, timedelta]] = None,
+    ):
+        self.type: AutomodActionType = enum_if_int(AutomodActionType, type)
+        self._metadata: AutomodActionMetadata = {}
+
+        # NOTE: if this is changed to do any sort of processing on those parameters,
+        # `_from_dict` would need to be updated as it doesn't pass any of them
+        if channel is not None:
+            self._metadata["channel_id"] = channel.id
+        if timeout_duration is not None:
+            if isinstance(timeout_duration, timedelta):
+                timeout_duration = int(timeout_duration.total_seconds())
+            self._metadata["duration_seconds"] = timeout_duration
+
+    @property
+    def channel_id(self) -> Optional[int]:
+        """Optional[:class:`int`]: The channel ID to send an alert in when the rule is triggered,
+        if :attr:`~AutomodAction.type` is :attr:`AutomodActionType.send_alert_message`."""
+        return _get_as_snowflake(self._metadata, "channel_id")
+
+    @property
+    def timeout_duration(self) -> Optional[int]:
+        """Optional[:class:`int`]: The duration (in seconds) for which to timeout
+        the user when the rule is triggered,
+        if :attr:`~AutomodAction.type` is :attr:`AutomodActionType.timeout`."""
+        return _get_as_snowflake(self._metadata, "duration_seconds")
+
+    def __repr__(self) -> str:
+        if self.type is AutomodActionType.send_alert_message:
+            meta_repr = f" channel_id={self.channel_id!r}"
+        elif self.type is AutomodActionType.timeout:
+            meta_repr = f" timeout_duration={self.timeout_duration}"
+        else:
+            meta_repr = ""
+        return f"<AutomodAction type={self.type!r}{meta_repr}>"
+
+    @classmethod
+    def _from_dict(cls, data: AutomodActionPayload) -> Self:
+        # bypass init overloads and unnecessary processing
+        self = cls.__new__(cls)
+
+        self.type = try_enum(AutomodActionType, data["type"])
+        self._metadata = data.get("metadata", {})
+
+        return self
+
+    def to_dict(self) -> AutomodActionPayload:
+        return {
+            "type": self.type.value,
+            "metadata": self._metadata,
+        }
+
+
+# TODO: perhaps this should just be a typeddict instead?
+class AutomodTriggerMetadata:
+    """
+    Metadata for an auto moderation trigger.
+
+    .. versionadded:: 2.6
```

Comment:
moved the documentation of these types into the correct section (`Data Classes` instead of `Discord Models`) in 0e24b075d49c74de2ad12e01dbbd6d00ba5c749d, I don't think this needs an additional wording change